package_policies/R8.md

```diff
@@ -0,0 +1 @@
+**R8.** Each xSDK member package should have sufficient documentation to support use and further development. Elements may include, for example, getting started, functionality documentation, on-ramping new developers, examples, and team processes. Sufficiency of documentation will be reviewed by the xSDK community.
```

Comment:
```suggestion
**R8.** Each xSDK member package should have sufficient documentation to support use and further development.
While it is difficult to formalize what constitutes sufficiency of documentation, the requirement will be reviewed by the
xSDK community based on documentation elements that may include, for example, the following:
  * Instructions on installation and getting started;
  * Documentation on functionality, for example class- and member-level documentation;
  * Discussions of how individual classes or groups of classes interact, i.e., at a level above individual classes but below example codes;
  * Presence of example codes that illustrate typical use cases;
  * Links to external resources such as mailing lists and forums where users and developers can find help;
  * Instructions for on-ramping new developers;
  * Description of team processes.
```

Reply:
+ Comparison of strengths and weaknesses relative to related software, to assist a prospective user in choosing appropriate technology.
+ Theory manual where appropriate, laying out the relevant mathematical abstractions and notation.
+ Low-commitment demos using technologies such as Binder or containers.

Reply:
I think @bangerth slightly longer version clarifies the quality metrics and helps the softwaredevelopers in deciding what documenttion to provide.

Reply:
Comments from the xSDK meeting on July 23:
We discussed Wolfgang's suggestion. Here are a few questions that came up:
Ideal: What is the "good enough" for a new user to get started?
Want to isolate the most important components for the mandatory policy (down the road). 
Instructions for installation and "Getting started" (this is still subjective). 
Careful that language isn't too C++ specific: E.g. member and class, maybe use 'API' instead.
Make language as generic as possible (Applicable to python, c, c++)

Reply:
Apologies for not being on the call yesterday -- I was on vacation this week, while also participating in two workshops; something had to give :-)

Do you want me to edit anything in my comments above, or did you come to agreement?

Reply:
We can have two sections, recommended and strongly recommended:
```
Strongly recommended:
* Instruction on installation and getting started, e.g., list of relevant CMake or Autoconf options
* Documentation of the front-facing (external) API, e.g., structs, classes, methods, etc.
* Provide a set of examples of common use-cases
Recommended:
* Documentation of the internal API, e.g., methods and classes that are not going to be directly called by the user
* Overview of the internal design describing the relations between different components
* Description of the pull-request policies, e.g., formatting and naming conventions, C++ standard, etc.
* Theory manual where appropriate, laying out the relevant mathematical abstractions and notation.
* Comparison of strengths and weaknesses relative to related software, to assist a prospective user in choosing appropriate technology.
* Low-commitment demos using technologies such as Binder or containers.
```
The last three are directly copied from @jedbrown 

Eventually the top three should be come mandatory.

Reply:
I am very happy with the current version. One comment maybe: I think the term "mailing list" should be understood in a broad sense and include modern versions of "mailing lists" such as Slack, Mattermost etc.

Reply:
Right. I think the proposed changes above, saying "mailing lists and forums" cover this.